### PR TITLE
Multi currency support (exchange rates only)

### PR DESCRIPTION
**Pull request by hannunosto**

### CODE REVIEW

File: classes/collections/exchange-rates.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+class NostoTaggingCollectionExchangeRates extends NostoExchangeRateCollection
+{
+    public function __construct($input, $flags=0, $iterator_class="ArrayIterator")
+    {
+        parent::__construct($input, $flags, $iterator_class);
+        
+        $this->dispatchHook(
+            'actionNostoExchangeRatesLoadAfter',
+            array(
+                'nosto_exhange_rates' => $this,
```
Comment (mridang) [inline classes/collections/exchange-rates.php]: Typo.

File: nostotagging.php [hunk 2]
```diff
@@ -109,6 +114,16 @@ class NostoTagging extends Module
...
             'description' => 'Action hook fired after a Nosto product object has been loaded.',
         ),
+        array(
+            'name' => 'actionNostoPriceVariantLoadAfter',
+            'title' => 'After load nosto price variation',
+            'description' => 'Action hook fired after a Nosto price variation object has been initialized.',
+        ),
+        array(
+            'name' => 'actionNostoRatesLoadAfter',
+            'title' => 'After load nosto exhange rates',
```
Comment (mridang) [inline nostotagging.php]: Typo

File: nostotagging.php [hunk 3]
```diff
@@ -109,6 +114,16 @@ class NostoTagging extends Module
...
+            'description' => 'Action hook fired after a Nosto exhange rate collection has been initialized.',
```
Comment (mridang) [inline nostotagging.php]: Typo

File: upgrade/upgrade-2.6.0.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+{
+    $hooks = array(
+        array(
+            'name' => 'actionNostoPriceVariantLoadAfter',
+            'title' => 'After load nosto price variation',
+            'description' => 'Action hook fired after a Nosto price variation object has been initialized.',
+        ),
+        array(
+            'name' => 'actionNostoRatesLoadAfter',
+            'title' => 'After load nosto exhange rates',
```
Comment (mridang) [inline upgrade/upgrade-2.6.0.php]: Typo.

File: upgrade/upgrade-2.6.0.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            'description' => 'Action hook fired after a Nosto exhange rate collection has been initialized.',
```
Comment (mridang) [inline upgrade/upgrade-2.6.0.php]: Typo.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,41 +1,43 @@
 # Change Log
 All notable changes to this project will be documented in this file.
-This project adheres to [Semantic Versioning](http://semver.org/).
+This project adheres to Semantic Versioning(http://semver.org/).
 
-## [Unreleased][unreleased]
 
-## [2.5.0]
+## 2.6.0
+- Add support for multiple currencies (using exhange rates)
```
Comment (mridang) [inline CHANGELOG.md]: Typo.

File: classes/helpers/config.php [hunk 7]
```diff
@@ -92,18 +96,18 @@ public function exists($name, $lang_id = null, $id_shop_group = null, $id_shop =
...
-        $config_lang_table = $config_table.'_lang';
+        $config_table = _DB_PREFIX_ . 'configuration';
+        $config_lang_table = $config_table . '_lang';
 
         Db::getInstance()->execute(
-            'DELETE `'.$config_lang_table.'` FROM `'.$config_lang_table.'`
-			LEFT JOIN `'.$config_table.'`
-			ON `'.$config_lang_table.'`.`id_configuration` = `'.$config_table.'`.`id_configuration`
-			WHERE `'.$config_table.'`.`name` LIKE "NOSTOTAGGING_%"'
+            'DELETE `' . $config_lang_table . '` FROM `' . $config_lang_table . '`
```
Comment (oostman) [inline classes/helpers/config.php]: Are the extra spaces around `.` on purpose?

File: views/templates/hook/header_embed-script.tpl [hunk 8]
```diff
@@ -28,7 +28,7 @@
     (function(){
         var sa = "{$server_address|escape:"javascript":"UTF-8"}";
         {literal}function a(a){var b,c,d=window.document.createElement("iframe");d.src="javascript:false",(d.frameElement||d).style.cssText="width: 0; height: 0; border: 0";var e=window.document.createElement("div");e.style.display="none";var f=window.document.createElement("div");e.appendChild(f),window.document.body.insertBefore(e,window.document.body.firstChild),f.appendChild(d);try{c=d.contentWindow.document}catch(g){b=document.domain,d.src="javascript:var d=document.open();d.domain='"+b+"';void(0);",c=d.contentWindow.document}return c.open()._l=function(){b&&(this.domain=b);var c=this.createElement("scr".concat("ipt"));c.src=a,this.body.appendChild(c)},c.write("<bo".concat('dy onload="document._l();">')),c.close(),d}var b="nostojs";window[b]=window[b]||function(a){(window[b].q=window[b].q||[]).push(a)},window[b].l=new Date;var c=function(d,e){if(!document.body)return setTimeout(function(){c(d,e)},30);e=e||{},window[b].o=e;var f=document.location.protocol,g=["https:"===f?f:"http:","//",e.host||sa,e.path||"/include/",d].join("");a(g)};window[b].init=c{/literal}
-    })();
+    })();;
```
Comment (oostman) [inline views/templates/hook/header_embed-script.tpl]: typo?

**Comment by oostman** [comment]
Looks good to me, left minor comments.

File: views/templates/hook/top_price_variation-tagging.tpl [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+* @author    Nosto Solutions Ltd <contact@nosto.com>
+* @copyright 2013-2016 Nosto Solutions Ltd
+* @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*}
+
+{if isset($nosto_price_variation) && $nosto_price_variation->getVariationId()}
+		<div class="nosto_variation" style="display:none">{$nosto_price_variation->getVariationId()|escape:'htmlall':'UTF-8'}</div>
```
Comment (mridang) [inline views/templates/hook/top_price_variation-tagging.tpl]: Indentation

File: views/templates/hook/footer-product_product-tagging.tpl [hunk 10]
```diff
@@ -57,8 +57,12 @@
...
 		{/foreach}
+			{if $nosto_product->getVariationId()}
+					<span class="variation_id">{$nosto_product->getVariationId()|escape:'htmlall':'UTF-8'}</span>
+			{/if}
 	</div>
-    {if isset($nosto_category) && is_object($nosto_category)}
-        <div class="nosto_category" style="display:none">{$nosto_category->category_string|escape:'htmlall':'UTF-8'}</div>
-    {/if}
+		{if isset($nosto_category) && is_object($nosto_category)}
+				<div class="nosto_category" style="display:none">{$nosto_category->category_string|escape:'htmlall':'UTF-8'}</div>
```
Comment (mridang) [inline views/templates/hook/footer-product_product-tagging.tpl]: Indentation

File: views/templates/hook/footer-product_product-tagging.tpl [hunk 11]
```diff
@@ -57,8 +57,12 @@
                 {/if}
             {/if}
 		{/foreach}
+			{if $nosto_product->getVariationId()}
+					<span class="variation_id">{$nosto_product->getVariationId()|escape:'htmlall':'UTF-8'}</span>
```
Comment (mridang) [inline views/templates/hook/footer-product_product-tagging.tpl]: Indentation

File: classes/helpers/config.php (hunk 7 above)
Comment (hannunosto) [inline classes/helpers/config.php]: Yep, that's the recommended way for concatenating strings in PHP. Makes it also easier to read.

**Pull request merged**